src/interface/types.rs

```diff
@@ -571,3 +611,43 @@ pub fn get_duration_from_millis(union_argument: Arg) ->Result<Option<interface::
 pub fn arg_nullity(union_argument: &Arg) -> bool {
     unsafe{union_argument.dispatch_cbuf}.is_null()
 }
+
+pub fn get_sigactionstruct<'a>(union_argument: Arg) -> Result<Option<&'a mut SigactionStruct>, i32> {
+    let pointer = unsafe{union_argument.dispatch_sigactionstruct};
+
+    if !pointer.is_null() {
```

Comment:
Could possibly put this if/else into a helper function since it is the same across all these new functions